### PR TITLE
feat: add functionality for add productos from product details

### PR DESCRIPTION
**Pull request by denis-gandel**

### CODE REVIEW

**Comment by JorgeHB69** [comment]
About User Experience:
There is a lack of usability if we don't show the button for add to cart, because:
* The user couldn't understand how to buy the product. (if not variant selected)
* We lost the "call to action"
* We constantly change the distribution of the items on the screen so that it could be confused.
* The user could not understand that we can't buy the product because the stock isn't enough.
image

Also, you aren't following the distribution established on mockups for the actions.
image

So, that's my advice about the experience. I think it is more understandable that the button is deactivated when there isn't enough stock or when no variants are selected. And provide enough feedback to the user to understand the stock status.

File: apps/merchant-web/src/components/ChipSelector.tsx [hunk 1]
```diff
@@ -1,33 +1,40 @@
...
-                let newIndex = index;
-                if (indexSelected == index) {
-                    newIndex = -1;
-                }
+export default function ChipSelector({
+  name,
+  options,
+  handleChange,
+}: Readonly<ChipSelectorProps>) {
+  const { selectedAttributes } = useShoppingItem();
```
Comment (JorgeHB69) [inline apps/merchant-web/src/components/ChipSelector.tsx]: The ChipSelector isn't used only on the product contexts; it is also used on the SortModal, so your changes cause the sorting fails on the catalog.

image

Please review that, and remember that the components should try to be abstract to be reused.

File: apps/merchant-web/src/components/ChipSelector.tsx [hunk 2]
```diff
@@ -1,33 +1,40 @@
-"use client"
-import {useState} from "react";
-import ChipSelectorStyle from "@/styles/general/ChipSelector.module.css"
+import { useShoppingItem } from "@/contexts/ShoppingItemContext";
+import ChipSelectorStyle from "@/styles/general/ChipSelector.module.css";
```
Comment (JorgeHB69) [inline apps/merchant-web/src/components/ChipSelector.tsx]: Based on the guidelines, the components should be marked to the client side with "use client"

**Review by JorgeHB69 — CHANGES_REQUESTED** [review]

**Comment by aguirrekarina** [comment]
https://github.com/user-attachments/assets/7113eae4-e392-499d-b090-a86deb5d2abd

The notifications for a product is displayed twice even though it is only clicked once

**Comment by aguirrekarina** [comment]
image

The functionality of the details of a product is not working from the home catalog.

**Comment by aguirrekarina** [comment]
image

As a recommendation, the button to delete the product, like the others, could be red.

File: apps/merchant-web/src/components/ChipSelector.tsx (hunk 1 above)
Comment (JorgeHB69) [reply apps/merchant-web/src/components/ChipSelector.tsx]: Your changes still affect to the sort feature.
Now the sort wasn't applied and the entire page was reload when I click on some chip

https://github.com/user-attachments/assets/41baf2c0-1d0c-4ab5-a42e-fc51f28f75de

**Comment by JorgeHB69** [comment]
> The functionality of the details of a product is not working from the home catalog.

In general product details is not working @denis-gandel

**Comment by JorgeHB69** [comment]
> > The functionality of the details of a product is not working from the home catalog.
> 
> In general product details is not working @denis-gandel

Looks like the id isn't provided to the context

The URL of the request is `http://localhost:5001/api/inventory/Product/undefined`
image

**Comment by denis-gandel** [comment]
> > > The functionality of the details of a product is not working from the home catalog.
> > 
> > 
> > In general product details is not working @denis-gandel
> 
> Looks like the id isn't provided to the context
> 
> The URL of the request is `http://localhost:5001/api/inventory/Product/undefined` image

Regarding this, we are using Axios, and we have a configuration where we already have the general url that would be the localhost:5001/api and so on, and only with that you should put the missing route

**Comment by aguirrekarina** [comment]
https://github.com/user-attachments/assets/c31af286-1477-471d-b7f8-13543c92fa8a

The notification is displayed for the second time when a product is already added to the cart, when you return to the home page.

**Comment by JorgeHB69** [comment]
On the shopping cart page is possible to add the quantity of one product that is already on the limit of the stock
image
(The stock quantity for this variant is 31)

**Review by JorgeHB69 — APPROVED** [review]

**Review by aguirrekarina — APPROVED** [review]

**Pull request merged**